tests/posix_semaphore/main.c

```diff
@@ -239,6 +239,10 @@ void test3(void)
 #ifdef BOARD_NATIVE
 /* native can sometime take more time to respond as it is not real time */
 #define TEST4_TIMEOUT_EXCEEDED_MARGIN (300)
+#elif CPU_FAM_NRF51
+/* nrf51 based boards needs a slightly higher margin value. Using 105us makes
+ test4 result more reliable. */
+#define TEST4_TIMEOUT_EXCEEDED_MARGIN (105)
 #else
 #define TEST4_TIMEOUT_EXCEEDED_MARGIN (100)
```

Comment:
can't we just increase this default to say `110` instead of having (another) board specific one? I mean `105` is not that much more than the default of `100`.

Reply:
Makes sense. Using 105 should be enough though. @cladmi reported that the microbit is always between 102 and 103 and I also have these results.

Reply:
Having currently a cpu specific variable shows that this value should come from somewhere else.

It is like the first part of TDD where you only define special cases handling until you reach a point where it can/should be put in common and you refactor to using the value from somewhere else.

Here I would assume it is a value that could even be interesting to have in an application and may be defined in timer area. Having these different cases shows that this value must be "board/cpu" specific and would need special handling.

So I would personally keep the special case.

Reply:
Thanks @cladmi. @smlng does it make sense to you ? if not, how do we proceed ?